File: 151004/Tishalovich/rest_basics/publisher/src/main/java/by/bsuir/dc/publisher/services/impl/mappers/LabelMapper.java
package by.bsuir.dc.publisher.services.impl.mappers;

import by.bsuir.dc.publisher.entities.dtos.request.LabelRequestTo;
import by.bsuir.dc.publisher.entities.dtos.response.LabelResponseTo;
import by.bsuir.dc.publisher.entities.Label;
import org.mapstruct.Mapper;

@Mapper
public interface LabelMapper
        extends EntityMapper<LabelRequestTo, LabelResponseTo, Label> {
}

File: 151001/Hlebanova/LR2/LR2/Mappers/LabelMapper.cs
using Riok.Mapperly.Abstractions;
using LR2.Dto.Request.CreateTo;
using LR2.Dto.Request.UpdateTo;
using LR2.Dto.Response;
using LR2.Models;

namespace LR2.Mappers;

[Mapper]
public static partial class LabelMapper
{
    public static partial Label Map(UpdateLabelRequestTo updateLabelRequestTo);
    public static partial Label Map(CreateLabelRequestTo createLabelRequestTo);
    public static partial LabelResponseTo Map(Label label);
    public static partial IEnumerable<LabelResponseTo> Map(IEnumerable<Label> labels);

    public static partial IEnumerable<Label> Map(
        IEnumerable<UpdateLabelRequestTo> labelRequestTos);
}